### PR TITLE
Improvements to dlna server

### PR DESCRIPTION
**Pull request by Bond-009**
* Improve response writer
* Add analyzers
* Error on warnings in release mode
* Disable doc warnings

### CODE REVIEW

File: Emby.Dlna/ContentDirectory/ControlHandler.cs [hunk 1]
```diff
@@ -237,101 +254,81 @@ public string GetValueOrDefault(IDictionary<string, string> sparams, string key,
...
                     }
                 }
-
-                writer.WriteFullEndElement();
-                //writer.WriteEndDocument();
             }
 
-            var resXML = builder.ToString();
+            xmlWriter.WriteFullEndElement();
+            xmlWriter.WriteFullEndElement();
```
Comment (JustAMan) [inline Emby.Dlna/ContentDirectory/ControlHandler.cs]: why twice?

File: Emby.Dlna/ContentDirectory/ControlHandler.cs [hunk 2]
```diff
@@ -237,101 +254,81 @@ public string GetValueOrDefault(IDictionary<string, string> sparams, string key,
...
-            return new[]
-                {
-                    new KeyValuePair<string,string>("Result", resXML),
-                    new KeyValuePair<string,string>("NumberReturned", provided.ToString(_usCulture)),
-                    new KeyValuePair<string,string>("TotalMatches", totalCount.ToString(_usCulture)),
-                    new KeyValuePair<string,string>("UpdateID", _systemUpdateId.ToString(_usCulture))
-                };
+            xmlWriter.WriteElementString("NumberReturned", provided.ToString(_usCulture));
+            xmlWriter.WriteElementString("TotalMatches", totalCount.ToString(_usCulture));
+            xmlWriter.WriteElementString("UpdateID", _systemUpdateId.ToString(_usCulture));
```
Comment (JustAMan) [inline Emby.Dlna/ContentDirectory/ControlHandler.cs]: why `_usCulture`?

File: Emby.Dlna/ContentDirectory/ControlHandler.cs [hunk 3]
```diff
@@ -354,99 +351,74 @@ public string GetValueOrDefault(IDictionary<string, string> sparams, string key,
...
+                    _didlBuilder.WriteItemElement(dlnaOptions, xmlWriter, i, _user, item, serverItem.StubType, deviceId, filter);
                 }
-
-                writer.WriteFullEndElement();
-                //writer.WriteEndDocument();
             }
 
-            var resXML = builder.ToString();
+            xmlWriter.WriteFullEndElement();
+            xmlWriter.WriteFullEndElement();
```
Comment (JustAMan) [inline Emby.Dlna/ContentDirectory/ControlHandler.cs]: again, why twice?

File: Emby.Dlna/ContentDirectory/ControlHandler.cs [hunk 4]
```diff
@@ -354,99 +351,74 @@ public string GetValueOrDefault(IDictionary<string, string> sparams, string key,
...
 
-            var sortOrders = new List<(string, SortOrder)>();
-            if (!folder.IsPreSorted)
+            var sortOrders = folder.IsPreSorted switch
             {
-                sortOrders.Add((ItemSortBy.SortName, sort.SortOrder));
-            }
+                true => Array.Empty<(string, SortOrder)>(),
+                false => new[] { (ItemSortBy.SortName, sort.SortOrder) }
+            };
```
Comment (JustAMan) [inline Emby.Dlna/ContentDirectory/ControlHandler.cs]: hmm, maybe a ternary would look nicer here?

**Comment by PrplHaz4** [comment]
As of https://github.com/jellyfin/jellyfin/pull/2326/commits/82112b67880265c34f2dcca1ee7f0acf52a156df this looks to have broken the BROWSE control again (so no media is listed when browsing via DLNA)...

**Comment by Bond-009** [comment]
I’ll take a look when I’ve got time (and maybe add some tests)

File: Emby.Dlna/ContentDirectory/ControlHandler.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Dlna/ContentDirectory/ControlHandler.cs]: We opened 2 elements, so we close 2

File: Emby.Dlna/ContentDirectory/ControlHandler.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Dlna/ContentDirectory/ControlHandler.cs]: Same as before 🤷‍♂

File: Emby.Dlna/ContentDirectory/ControlHandler.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Dlna/ContentDirectory/ControlHandler.cs]: 2 open elements

**Comment by Bond-009** [comment]
Don't know why it's broken atm

**Comment by Bond-009** [comment]
@PrplHaz4 Did that last commit fix it?

**Comment by PrplHaz4** [comment]
> 
> 
> @PrplHaz4 Did that last commit fix it?

@Bond-009 At first blush it does not appear to have fixed it. I will try to grab some wireshark traces.

On that note, have you happened to look at why the DLNA debug logging isn't working? (#2317) or is that on purpose since they're so chatty?

**Comment by PrplHaz4** [comment]
@Bond-009 Two things I noticed that differ between the working responses and the non-working ones:
1. DIDL-Lite contents of <Result> is HTML Encoded in the working (old) response but not the new one
` <u:Result>&lt;DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"&gt;
&lt;container restricted="1" searchable="1" childCount="582" id="2b2bca16aacc8a14d53a11bb829eafa5" parentID="e9d5075a555c1cbc394eec4cef295274"&gt;`...etc

2. Count of records returned is off by one - `<NumberReturned>` is 3, but there are only 2 `<container>`s returned

I will keep looking but hopefully this is enough to go on.

**Comment by Bond-009** [comment]
@PrplHaz4 Can you give the full responses with and without the changes?

**Comment by PrplHaz4** [comment]
> 
> 
> @PrplHaz4 Can you give the full responses with and without the changes?

BEFORE (master 1ed92b05fb19a75f0fbc02def246dbc068687488)
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Envelope SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
	<Body>
		<BrowseResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
			<Result>
				&lt;DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"&gt;&lt;container restricted="1" searchable="1" childCount="129" id="2b2bca16aacc8a14d53a11bb829eafa5" parentID="e9d5075a555c1cbc394eec4cef295274"&gt;&lt;dc:title&gt;Live TV&lt;/dc:title&gt;&lt;upnp:class&gt;object.container.storageFolder&lt;/upnp:class&gt;&lt;/container&gt;&lt;container restricted="1" searchable="1" childCount="6" id="f137a2dd21bbc1b99aa5c0f6bf02a805" parentID="e9d5075a555c1cbc394eec4cef295274"&gt;&lt;dc:title&gt;Movies&lt;/dc:title&gt;&lt;upnp:class&gt;object.container.storageFolder&lt;/upnp:class&gt;&lt;upnp:albumArtURI dlna:profileID="JPEG_SM"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/480/480/0/0&lt;/upnp:albumArtURI&gt;&lt;upnp:icon&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/48/48/0/0&lt;/upnp:icon&gt;&lt;res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/4096/4096/0/0&lt;/res&gt;&lt;res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_MED;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/1024/768/0/0&lt;/res&gt;&lt;res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_SM;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="640x360"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/640/480/0/0&lt;/res&gt;&lt;res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/4096/4096/0/0&lt;/res&gt;&lt;res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/160/160/0/0&lt;/res&gt;&lt;res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90"&gt;http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/160/160/0/0&lt;/res&gt;&lt;/container&gt;&lt;/DIDL-Lite&gt;
			</Result>
			<NumberReturned>
				2
			</NumberReturned>
			<TotalMatches>
				2
			</TotalMatches>
			<UpdateID>
				2063
			</UpdateID>
		</BrowseResponse>
	</Body>
</Envelope>
```

AFTER (dlnawarnings https://github.com/jellyfin/jellyfin/pull/2326/commits/bade35958aadd6992b276aca278459ef5562d4ae)
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Envelope SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
	<Body>
		<BrowseResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
			<Result>
				<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/">
					<container restricted="1" searchable="1" childCount="129" id="2b2bca16aacc8a14d53a11bb829eafa5" parentID="e9d5075a555c1cbc394eec4cef295274">
						<title>
							Live TV
						</title>
						<class>
							object.container.storageFolder
						</class>
					</container>
					<container restricted="1" searchable="1" childCount="6" id="f137a2dd21bbc1b99aa5c0f6bf02a805" parentID="e9d5075a555c1cbc394eec4cef295274">
						<title>
							Movies
						</title>
						<class>
							object.container.storageFolder
						</class>
						<albumArtURI dlna:profileID="JPEG_SM">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/480/480/0/0
						</albumArtURI>
						<icon>
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/48/48/0/0
						</icon>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/4096/4096/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_MED;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/1024/768/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_SM;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="640x360">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/640/480/0/0
						</res>
						<res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/4096/4096/0/0
						</res>
						<res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/160/160/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/160/160/0/0
						</res>
					</container>
				</DIDL-Lite>
			</Result>
			<NumberReturned>
				2
			</NumberReturned>
			<TotalMatches>
				2
			</TotalMatches>
			<UpdateID>
				2063
			</UpdateID>
		</BrowseResponse>
	</Body>
</Envelope>
```

**Comment by Bond-009** [comment]
I see, I’ll fix this tonight

**Comment by Bond-009** [comment]
@PrplHaz4 What filter are you using on wireshark to capture these, so I can stop bothering you ;)

**Comment by PrplHaz4** [comment]
@Bond-009 doesn't seem to be fixed with https://github.com/jellyfin/jellyfin/pull/2326/commits/fc578de3b3370676090da5063faaf126ea50d2ec
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Envelope SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
	<Body>
		<BrowseResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
			<Result>
				<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/">
					<container restricted="1" searchable="1" childCount="89" id="2b2bca16aacc8a14d53a11bb829eafa5" parentID="e9d5075a555c1cbc394eec4cef295274">
						<title>
							Live TV
						</title>
						<class>
							object.container.storageFolder
						</class>
					</container>
					<container restricted="1" searchable="1" childCount="6" id="f137a2dd21bbc1b99aa5c0f6bf02a805" parentID="e9d5075a555c1cbc394eec4cef295274">
						<title>
							Movies
						</title>
						<class>
							object.container.storageFolder
						</class>
						<albumArtURI dlna:profileID="JPEG_SM">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/480/480/0/0
						</albumArtURI>
						<icon>
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/48/48/0/0
						</icon>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/4096/4096/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_MED;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/1024/768/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_SM;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="640x360">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/640/480/0/0
						</res>
						<res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_LRG;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="960x540">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/4096/4096/0/0
						</res>
						<res protocolInfo="http-get:*:image/png:DLNA.ORG_PN=PNG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/png/160/160/0/0
						</res>
						<res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=00;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00D00000000000000000000000000000" resolution="160x90">
							http://192.168.4.14:8096/Items/f137a2dd21bbc1b99aa5c0f6bf02a805/Images/Primary/0/985d4292b159647f2eac85ee1b6b5489/jpg/160/160/0/0
						</res>
					</container>
				</DIDL-Lite>
			</Result>
			<NumberReturned>
				2
			</NumberReturned>
			<TotalMatches>
				2
			</TotalMatches>
			<UpdateID>
				2069
			</UpdateID>
		</BrowseResponse>
	</Body>
</Envelope>
```

**Comment by PrplHaz4** [comment]
> 
> 
> @PrplHaz4 What filter are you using on wireshark to capture these, so I can stop bothering you ;)

@Bond-009 haha...you have too much faith in me...
search box: `ip.src_host contains "jftest" || ip.dst_host contains "jftest"`
and then just look for `POST /dlna/af9f06335f4d490685987391b5f6b4be/contentdirectory/control HTTP/1.1` and the response that goes with it 🤷‍♂

**Comment by Bond-009** [comment]
VLC can't seem to find my instance :/

**Comment by PrplHaz4** [comment]
It'll be a couple hours before I can test again. Maybe check your dlna settings I'm jf and make sure intervals are 60 and 120 instead of the 1800 that was default at one point.

**Comment by PrplHaz4** [comment]
@Bond-009 nope, not that one https://github.com/jellyfin/jellyfin/pull/2326/commits/6b68521bae63ecd2c8dd3d5a912b70756c1c5e49

```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Envelope SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
	<Body>
		<Fault>
			<faultcode>
				500
			</faultcode>
			<faultstring>
				WriteStartDocument cannot be called on writers created with ConformanceLevel.Fragment.
			</faultstring>
			<detail>
				<UPnPError xmlns="urn:schemas-upnp-org:control-1-0">
					<errorCode>
						401
					</errorCode>
					<errorDescription>
						Invalid Action
					</errorDescription>
				</UPnPError>
			</detail>
		</Fault>
	</Body>
</Envelope>
```

**Comment by cvium** [comment]
> Note that since the DIDL-Lite format of Result is based on XML, it needs to be escaped (using the normal XML rules: [XML] Section 2.4 Character Data and Markup) before embedding in a SOAP response message. In addition, when a variable of type A_ARG_TYPE_Result is employed as a parameter in a comma-separated list, commas (‘,’) that appear within XML CDATA must be escaped (as ,), see section 2.3.1.

Section 2.5.4 http://upnp.org/specs/av/UPnP-av-ContentDirectory-v1-Service.pdf

**Comment by PrplHaz4** [comment]
@Bond-009 doesn't look like https://github.com/jellyfin/jellyfin/pull/2326/commits/36d62c30930fa5b536fa5c7d1f4c85243a4b3fea is working but didn't have time to grab a trace on the train - will have to get it tonight.

**Comment by Bond-009** [comment]
@PrplHaz4 My bad, forgot to revert the previous (failed) fix

**Comment by PrplHaz4** [comment]
> 
> 
> @PrplHaz4 My bad, forgot to revert the previous (failed) fix

@Bond-009 not working as of https://github.com/jellyfin/jellyfin/pull/2326/commits/6ee04855cd112255d4b73b24cb5511a41c1db1f9 - will be a few hrs before I can get a trace - maybe time to fix logging in the DLNA area - it looks like the control requests aren't being logged anymore either.  VLC and Windows Explorer are not listing any media under jellyfin.

**Comment by PrplHaz4** [comment]
@Bond-009 Here's the response with https://github.com/jellyfin/jellyfin/pull/2326/commits/6ee04855cd112255d4b73b24cb5511a41c1db1f9 - "Result" is empty

```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Envelope SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
	<Body>
		<BrowseResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
			<Result />
			<NumberReturned>
				2
			</NumberReturned>
			<TotalMatches>
				2
			</TotalMatches>
			<UpdateID>
				2050
			</UpdateID>
		</BrowseResponse>
	</Body>
</Envelope>
```

**Comment by PrplHaz4** [comment]
Looks like that https://github.com/jellyfin/jellyfin/pull/2326/commits/d69870680b9a12118b2e5b5e9944c91b886a12c9 did it!  Thanks and nice work @Bond-009

Movies are working fine now, but VLC is not listing my LiveTV channels through DLNA.  The individual channels are listed on my Samsung TV DLNA client and Windows Explorer.  That might be another issue entirely - LMK if you think it's worth opening a new issue with new traces...
image

**Comment by PrplHaz4** [comment]
> 
> 
> Looks like that [d698706](https://github.com/jellyfin/jellyfin/commit/d69870680b9a12118b2e5b5e9944c91b886a12c9) did it! Thanks and nice work @Bond-009
> 
> Movies are working fine now, but VLC is not listing my LiveTV channels through DLNA. The individual channels are listed on my Samsung TV DLNA client and Windows Explorer. That might be another issue entirely - LMK if you think it's worth opening a new issue with new traces...

@Bond-009 answering my own question - this PR should be merged and I will create a new issue for the LiveTV Browse in VLC over DLNA.

**Comment by cvium** [comment]
Hmm, I forgot that Github creates merge commits when handling conflicts.

**Comment by PrplHaz4** [comment]
> 
> 
> > Looks like that [d698706](https://github.com/jellyfin/jellyfin/commit/d69870680b9a12118b2e5b5e9944c91b886a12c9) did it! Thanks and nice work @Bond-009
> > Movies are working fine now, but VLC is not listing my LiveTV channels through DLNA. The individual channels are listed on my Samsung TV DLNA client and Windows Explorer. That might be another issue entirely - LMK if you think it's worth opening a new issue with new traces...
> 
> @Bond-009 answering my own question - this PR should be merged and I will create a new issue for the LiveTV Browse in VLC over DLNA.

As promised, here is the issue I've created for LiveTV/DLNA/VLC: #2371

File: Emby.Dlna/PlayTo/SsdpHttpClient.cs [hunk 5]
```diff
@@ -90,7 +93,7 @@ private static string NormalizeServiceUrl(string baseUrl, string serviceUrl)
             options.RequestHeaders["NT"] = "upnp:event";
             options.RequestHeaders["TIMEOUT"] = "Second-" + timeOut.ToString(_usCulture);
 
-            using (await _httpClient.SendAsync(options, "SUBSCRIBE").ConfigureAwait(false))
+            using (await _httpClient.SendAsync(options, new HttpMethod("SUBSCRIBE")).ConfigureAwait(false))
```
Comment (dkanada) [inline Emby.Dlna/PlayTo/SsdpHttpClient.cs]: Does `HttpMethod.Subscribe` exist?

Comment (Bond-009) [reply Emby.Dlna/PlayTo/SsdpHttpClient.cs]: Nope

**Pull request merged**